app/src/main/java/tech/jpco/qen/view/MainActivity.kt

```diff
@@ -0,0 +1,138 @@
+package tech.jpco.qen.view
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.widget.Button
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProviders
+import com.jakewharton.rxbinding3.view.clicks
+import com.jakewharton.rxbinding3.view.layoutChanges
+import io.reactivex.Completable
+import io.reactivex.Observable
+import io.reactivex.Scheduler
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+import kotlinx.android.synthetic.main.activity_main.*
+import tech.jpco.qen.R
+import tech.jpco.qen.iLogger
+import tech.jpco.qen.log
+import tech.jpco.qen.model.MyPagesRepository
+import tech.jpco.qen.viewModel.MetaEvent
+import tech.jpco.qen.viewModel.QenViewModel
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
+class MainActivity : AppCompatActivity() {
+    private val cd = CompositeDisposable()
+    private val vm by lazy { ViewModelProviders.of(this).get(QenViewModel::class.java) }
+
+    @SuppressLint("CheckResult")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        setupInStreams()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        setupOutStreams()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        iLogger("destroying")
+        cd.clear()
+    }
+
+    private fun setupOutStreams() {
+        fun <T : MetaEvent> Button.throttledMetaEvent(
+            output: T,
+            minMillis: Long = 500
+        ): Observable<T> =
+            this.clicks()
+                .throttleFirst(minMillis, TimeUnit.MILLISECONDS)
+                .map { output }
+
+
+        val metaStream: Observable<MetaEvent> =
+            qenPage.arStream.doOnNext { iLogger("AR", it) }
+                .publish { aspectRatios ->
+                    Observable.merge(
+                        aspectRatios.startWith(0f).switchMap {
+                            new_button.throttledMetaEvent(
+                                MetaEvent.NewPage(
+                                    it
+                                )
+                            )
+                        },
+                        aspectRatios.take(1).map { MetaEvent.CurrentPage(it) },
+                        cycle_button.throttledMetaEvent(MetaEvent.CyclePage, 50),
+                        clear_button.throttledMetaEvent(MetaEvent.UiClearPage)
+
+                    )
+                }
+                .doOnNext { iLogger("MetaEvent emitted new", it) }
+                .replay(500, TimeUnit.MILLISECONDS)
+                .apply {
+                    connect().also {
```

Comment:
I think you can collapse this `apply { connect().also {` into just `connect().also {`